internal/build/docker.go

```diff
@@ -54,7 +58,42 @@ func (l *localDockerBuilder) BuildDocker(ctx context.Context, baseDockerfile str
 	return baseTag, nil
 }
 
-func (l *localDockerBuilder) buildBaseWithMounts(ctx context.Context, baseDockerfile string, tag string, mounts []Mount) (digest.Digest, error) {
+// Naively tag the digest and push it up to the docker registry specified in the name.
+//
+// TODO(nick) In the future, I would like us to be smarter about checking if the kubernetes cluster
+// we're running in has access to the given registry. And if it doesn't, we should either emit an
+// error, or push to a registry that kubernetes does have access to (e.g., a local registry).
+func (l *localDockerBuilder) PushDocker(ctx context.Context, name string, dig digest.Digest) error {
+	ref, err := reference.ParseNormalizedNamed(name)
+	if err != nil {
+		return fmt.Errorf("PushDocker: %v", err)
+	}
+
+	if reference.Domain(ref) == "" {
+		return fmt.Errorf("PushDocker: no domain in container name: %s", ref)
+	}
+
+	tag, err := reference.WithTag(ref, pushTag)
+	if err != nil {
+		return fmt.Errorf("PushDocker: %v", err)
+	}
+
+	err = l.dcli.ImageTag(ctx, dig.String(), tag.String())
+	if err != nil {
+		return fmt.Errorf("PushDocker#ImageTag: %v", err)
+	}
+
+	outputIO, err := l.dcli.ImagePush(
```

Comment:
lgtm! tho can you explain the difference between `ImageTag` and `ImagePush`? docs for `ImageTag` say it tags the image "in the host"...?